be/src/storage/lake/vacuum.cpp

```diff
@@ -640,15 +640,19 @@ void delete_txn_log(TabletManager* tablet_mgr, const DeleteTxnLogRequest& reques
     DCHECK(response != nullptr);
 
     std::vector<std::string> files_to_delete;
-    files_to_delete.reserve(request.tablet_ids_size() * request.txn_ids_size());
+    files_to_delete.reserve(request.tablet_ids_size() * (request.txn_ids_size() + request.txn_infos_size()));
 
     for (auto tablet_id : request.tablet_ids()) {
         for (auto txn_id : request.txn_ids()) {
             auto log_path = tablet_mgr->txn_log_location(tablet_id, txn_id);
             files_to_delete.emplace_back(log_path);
-
             tablet_mgr->metacache()->erase(log_path);
         }
+        for (auto&& info : request.txn_infos()) {
```

Comment:
this implementation looks like both txn_ids and txn_infos can be set in the same request. better add comments to indicate that this is a just implementation, FE won't set both in a single request.